components/ListHeader/list-header.tsx

```diff
@@ -42,11 +44,13 @@ const ListHeader = ({ list, listId, isOwner }: ListHeaderProps): JSX.Element =>
         <div className="flex flex-col justify-center p-2 header-info grow">
           <div className="flex gap-2">
             <Title level={1} className="!text-2xl font-semibold tracking-tight text-slate-900">
-              {(list && truncateString(list.name, 30)) || "Insights"}
+              {(name && truncateString(name, 30)) || "Insights"}
```

Comment:
nit: (non blocking) just a carry-over
```suggestion
              {(name && truncateString(name, 30)) || "List"}
```